packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts

```diff
@@ -0,0 +1,337 @@
+		if ( !this._wrapper ) {
+			return;
+		}
+
+		this._wrapper.remove();
+		this._wrapper = null;
+
+		this.editor.editing.view.focus();
+
+		this.refresh();
+	}
+
+	/**
+	 * Save edited image. In case server respond with "success" replace with edited image,
+	 * otherwise show notification error.
+	 */
+	private _handleImageEditorOnSave( state: ProcessingState, asset: CKBoxRawAssetDefinition ) {
+		const t = this.editor.locale.t;
+		const notification = this.editor.plugins.get( Notification );
+		const pendingActions = this.editor.plugins.get( PendingActions );
+		const action = pendingActions.add( t( 'Processing the edited image.' ) );
+
+		this._processInProgress.set( state.ckboxImageId, state );
+		this._showImageProcessingIndicator( asset );
+
+		this._waitForAssetProcessed( asset.data.id, state.controller.signal )
+			.then(
+				asset => {
+					this._replaceImage( state.element, asset );
+				},
+				() => {
+					if ( !state.controller.signal.aborted ) {
+						notification.showWarning( t( 'Server failed image processing or didn\'t respond' ), {
+							title: t( 'Server failed image processing or didn\'t respond' ),
+							namespace: 'ckbox'
+						} );
+					}
+
+					// Remove processing indicator. It was added only to ViewElement.
+					this.editor.editing.reconvertItem( state.element );
+				}
+			).finally( () => {
+				this._processInProgress.delete( state.ckboxImageId );
+				pendingActions.remove( action );
+			} );
+
+		this.refresh();
+	}
+
+	/**
+	 * Get asset's status on server. If server responds with "success" status then
+	 * image is already proceeded and ready for saving.
+	 */
+	private async _getAssetStatusFromServer( id: string, signal: AbortSignal ): Promise<CKBoxRawAssetDefinition> {
+		const url = new URL( 'assets/' + id, this.editor.config.get( 'ckbox.serviceOrigin' )! );
+		const ckboxEditing = this.editor.plugins.get( CKBoxEditing );
+		const response: CKBoxRawAssetDataDefinition = await sendHttpRequest( {
+			url,
+			signal,
+			authorization: ckboxEditing.getToken().value
+		} );
+		const status = response.metadata!.metadataProcessingStatus;
+
+		if ( !status || status == 'queued' ) {
+			throw new Error( 'Image has not been processed yet.' );
+		}
+
+		return { data: { ...response } };
+	}
+
+	private async _waitForAssetProcessed( id: string, signal: AbortSignal ): Promise<CKBoxRawAssetDefinition> {
+		const result = await retry(
+			() => this._getAssetStatusFromServer( id, signal ),
+			{
+				signal,
+				maxAttempts: 5
+			}
+		);
+
+		if ( result.data.metadata!.metadataProcessingStatus != 'success' ) {
+			throw new Error( 'Image processing failed.' );
+		}
+
+		return result;
+	}
+
+	/**
+	 * Shows processing indicator while image is processing.
+	 *
+	 * @param asset Data about certain asset.
+	 */
+	private _showImageProcessingIndicator( asset: CKBoxRawAssetDefinition ): void {
+		const editor = this.editor;
+		const selectedImageElement = editor.model.document.selection.getSelectedElement()!;
```

Comment:
Please use the `element` from `state`.

Reply:
I used it, and unfortunately, it caused some tests to fail (`ckboximageedit/ckboximageeditcommand`).
<img width="876" alt="Screenshot 2023-11-20 at 16 44 14" src="https://github.com/ckeditor/ckeditor5/assets/26603574/5149f701-31f6-4e3b-9126-750e43e7e8e7">

Reply:
Update: I was able to make the changes and fix the tests.